3rdparty/zlib-ng/CMakeLists.txt

```diff
@@ -789,8 +789,4 @@ if(ENABLE_SOLUTION_FOLDERS)
   set_target_properties(${ZLIB_LIBRARY} PROPERTIES FOLDER "3rdparty")
 endif()
 
-if(NOT BUILD_SHARED_LIBS)
-  ocv_install_target(${ZLIB_LIBRARY} EXPORT OpenCVModules ARCHIVE DESTINATION ${OPENCV_3P_LIB_INSTALL_PATH} COMPONENT dev)
-endif()
```

Comment:
It should break static linkage, if OpenCV is build against own zlib-ng, but not system-wide.

Reply:
Sorry for late reply. This `ocv_install_target` statement isn't removed. It is moved to outer space because both static linkage and dynamic linkage need this statement in Android build. So, this will only affect dynamic linkage.

Reply:
https://github.com/opencv/ci-gha-workflow/blob/main/.github/workflows/OCV-PR-4.x-Android-Test.yaml
However, I cannot find previous build failure log.